Document join-pushdown.enabled for JDBC connectors

hashhar opened the pull request:
I did not document the session property `join_pushdown_enabled` since I couldn't find examples of session properties being documented for JDBC connectors.

If needed I can add that too provided someone can point me to an existing example for the convention.

mosabua commented on docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:
```diff
@@ -33,3 +33,10 @@ connector:
       negatively impact performance.
     - 1000
+  * - ``join-pushdown.enabled``
+    - Enable join pushdown. Join pushdown is disabled by default as this is the
```
Link to the join pushdown section in the pushdown docs.

mosabua replied:
by default.

jhlodin replied:
> Link to the join pushdown section in the pushdown docs.

Can do this with 

``` Enable :ref:`join_pushdown` ```

mosabua reviewed (changes requested):
Add the catalog session property in the description text. There is no standard text. But you could use something like ..

The corresponding catalog session property is xyz.

hashhar commented on docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:
```diff
@@ -33,3 +33,10 @@ connector:
     - 1000
+  * - ``join-pushdown.enabled``
+    - Enable join pushdown. Join pushdown is disabled by default as this is the
+      safer option. Pushing down a join which substantially increases the row
```
Pushdown where the data transfer is larger is worse than no pushdown case (unless your database lives on localhost).

A default that has the possibility of making performance a lot worse is not a good default. See for example the parquet column index support that was added but enabled by default which led to regressions on some cases (which as it turns out are pretty common and widespread among Trino users).

I don't agree that the default should be changed.

mosabua replied:
I was speaking hypothetically based on chats about other pushdown with @electrum  .. I am totally fine with leaving the existing default.

We probably should reword the explanation then ... let me know if you want to reword or get something from me.

hashhar replied:
I've opted to simplify it. PTAL.

jhlodin commented on docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:
```diff
@@ -33,3 +33,10 @@ connector:
+  * - ``join-pushdown.enabled``
+    - Enable join pushdown. Join pushdown is disabled by default as this is the
+      safer option. Pushing down a join which substantially increases the row
+      count vs sizes of left and right table separately, may incur huge cost
+      both in terms of performance and money due to an increased network
+      traffic.
```
Like the `write.batch-size` property above, we probably don't need to give much more detail than "can negatively impact performance" or similar.

hashhar replied:
Fair point. The information isn't actionable anyway since the only option is "yes" or "no". 👍

hashhar commented on docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:
```diff
@@ -33,3 +33,7 @@ connector:
     - 1000
+  * - ``join-pushdown.enabled``
+    - Enable :ref:`join pushdown <join-pushdown>`. Join pushdown is disabled by
+      default. Enabling this may negatively impact performance for some queries.
```
Should I add this?

```rst
It can also be enabled using by setting the :doc:`catalog session property </sql/set-session>` ``join_pushdown_enabled`` to ``true``.
```

jhlodin replied:
Would condense into "Can also be enabled on a per-session basis with the ``join_pushdown_enabled`` :doc:`session property </sql/set-session>`.

hashhar commented:
AC and squashed.

mosabua commented on docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:
```diff
@@ -33,3 +33,9 @@ connector:
+    - Enable :ref:`join pushdown <join-pushdown>`. Join pushdown is disabled by
+      default. Enabling this may negatively impact performance for some queries.
+      It can also be enabled on a per-session basis with the
+      ``join_pushdown_enabled`` :doc:`session property </sql/set-session>`.
```
catalog session property

hashhar replied:
Done.

hashhar commented:
![image](https://user-images.githubusercontent.com/10473142/134622843-4990cc43-5331-4f79-8be7-82f84fc44bba.png)

I think I've addressed all comments now. PTAL @mosabua if this looks good to go.

jhlodin commented on docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:
```diff
@@ -33,3 +33,10 @@ connector:
       negatively impact performance.
     - 1000
+  * - ``join-pushdown.enabled``
+    - Enable :ref:`join pushdown <join-pushdown>`. Join pushdown is disabled by
+      default. Enabling this may negatively impact performance for some queries.
```
Would suggest removing "Join pushdown is disabled by default" since it's saying the same thing as showing the default value to be "false".

jhlodin reviewed (approved):
Couple minor comments, otherwise LGTM

jhlodin commented:
Looks good! FYI Manfred is out today and has spotty availability next week, up to you if you'd like to wait for his review.

mosabua commented on docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:
```diff
@@ -33,3 +33,9 @@ connector:
       negatively impact performance.
     - 1000
+  * - ``join-pushdown.enabled``
+    - Enable :ref:`join pushdown <join-pushdown>`. Can also be enabled on a
```
Can we simplify this sentence maybe just

```
Equivalent catalog session property is ``join_pushdown_enabled``.
```

mosabua replied:
and you can optional link but not even sure that is necessary.

mosabua reviewed (approved):
Ship it!
